### PR TITLE
[Ready for review] Adjusting instructions for local builds in two different files

### LINKED ISSUE #3368: [BUG] Misaligned instructions for local builds in two different files
### Description
The [README inside of the `book` directory](https://github.com/the-turing-way/the-turing-way/tree/main/book) has different instructions for how to build the book locally, compared to the [Community Handbook instructions](https://the-turing-way.netlify.app/community-handbook/local-build).

#### Expected instructions
- [ ] Install miniconda
- [ ] Create new environment
- [ ] Clone repository
- [ ] Install requirements
- [ ] Build the book

#### Actual instructions (in [`book/README.md`](https://github.com/the-turing-way/the-turing-way/tree/main/book))
- [ ] Install requirements
- [ ] Build the book

#### Solution
Make a decision, to either
- (option 1) copy the text from the [Community Handbook instructions](https://the-turing-way.netlify.app/community-handbook/local-build) to `book/README.md`, or
- (option 2) remove the instructions completely in [`book/README.md`](https://github.com/the-turing-way/the-turing-way/tree/main/book) and replace them with a link to the [Community Handbook instructions](https://the-turing-way.netlify.app/community-handbook/local-build).

### PR DESCRIPTION
**Pull request by kallewesterling**
There was a problem discussed in #3368 where there were different instructions for how to deploy the book locally. These revision addresses that problem by:

- [X] moving missing text from README to the local-build.md file
- [X] cleaning up the markdown
- [X] adding note in README to look for local deploy instructions in TTW

Resolves #3368.

### CODE REVIEW

File: book/website/community-handbook/local-build.md [hunk 1]
```diff
@@ -47,33 +72,64 @@ You will need to locate your "terminal" or "prompt" application on your machine.
...
+
+When you test your edits by building the book multiple times, it is better to clean up the last build before generating a new one.
+You can either manually delete the `book/website/_build` folder every time, or run this command:
+
+```bash
+cd book/website
+jupyter-book clean .
+```
+
+More details on this process can be read on the [Jupyter Book's GitHub repository](https://github.com/executablebooks/jupyter-book/blob/master/docs/advanced/advanced.md#clean-your-books-generated-files).
```
Comment (JimMadge) [inline book/website/community-handbook/local-build.md:97]:
```suggestion
More details on this process can be read in [Jupyter Book's documentation](https://jupyterbook.org/en/stable/basics/build.html?highlight=clean#clean-your-books-generated-files).
```

File: book/website/community-handbook/local-build.md [hunk 2]
```diff
@@ -47,33 +72,64 @@ You will need to locate your "terminal" or "prompt" application on your machine.
...
+
+
+### Check external links in the book
+
+When editing or reviewing this book locally, you can run the Sphinx link checker with Jupyter Book to check if the external links mentioned in the book are valid.
+To run the link checker, use the following command:
+
+```bash
```
Comment (JimMadge) [inline book/website/community-handbook/local-build.md:105]:
```suggestion
```console
```

File: book/website/community-handbook/local-build.md [hunk 3]
```diff
@@ -47,33 +72,64 @@ You will need to locate your "terminal" or "prompt" application on your machine.
...
+cd book/website
+jupyter-book build . --builder linkcheck
+```
+
+The link checker checks if the each link resolves and prints the status on your terminal so that you can check and resolve any incorrect links.
+Read more about this on the [Jupyter Book's GitHub repository](https://github.com/executablebooks/jupyter-book/blob/master/docs/advanced/advanced.md#check-external-links-in-your-book).
```
Comment (JimMadge) [inline book/website/community-handbook/local-build.md:111]:
```suggestion
Read more about this in [Jupyter Book's documentation](https://jupyterbook.org/en/stable/advanced/html.html?highlight=check%20external#check-external-links-in-your-book).
```

File: book/website/community-handbook/local-build.md [hunk 4]
```diff
@@ -47,33 +72,64 @@ You will need to locate your "terminal" or "prompt" application on your machine.
...
 
-But this can be difficult! Even with an organized, concerted effort, package management for programming languages naturally throws up dependency issues. Python packages, for reasons not discussed here, tend to suffer from dependency issues a bit more than other languages (note that all languages do!) and one guaranteed way to come across such an issue by trying to maintain all of your Python projects using just one, large set of packages, each at a specific version. You simply can't cater to the needs of all package dependencies this way. https://xkcd.com/1987/   
+But this can be difficult! Even with an organized, concerted effort, package management for programming languages naturally throws up dependency issues. Python packages, for reasons not discussed here, tend to suffer from dependency issues a bit more than other languages (note that all languages do!) and one guaranteed way to come across such an issue by trying to maintain all of your Python projects using just one, large set of packages, each at a specific version. You simply can't cater to the needs of all package dependencies this way.
+
 ![](https://imgs.xkcd.com/comics/python_environment.png)
```
Comment (JimMadge) [inline book/website/community-handbook/local-build.md:121]: I think we are falling short of the CC BY-NC requirements here. We need to provide attribution and a link to the licence.

The image also has no caption or alt-text.

I think we could kill two birds with one stone here by adding a caption with a link to the xkcd page (which should be sufficient for attribution, showing changes and licence). We should also have alt-text explaining the image. Might be worth asking @the-turing-way/accessibility for advice.

Comment (kallewesterling) [reply book/website/community-handbook/local-build.md:121]: Absolutely! I forgot to mention that I did see this when reviewing the page, but thought that's a separate issue than the one I'm addressing in this PR, but if you think we should do it here anyway, I can try to do something to this effect!

**Comment by sgibson91** [comment]
> I'm not a fan on Conda so that isn't the set of instructions I would choose. It also _isn't_ how the book actually gets built for deployment. If we want portability, I would go with something like pip-tools/hatch (which I think handles 99% of the Python packaging challenges) and a container.

I think we definitely want portability across _systems_ (specifically, CI and publishing sites), but I'm not sure that we want to be so prescriptive about how people build the book locally in order to contribute, especially at the level of adding Markdown files. A container is already pretty beefy technology given this project's contributor base, whereas I've had to mentor people through just installing Python to get the book to build previously. There's also no _requirement_ to build the book locally in the same way that contributing to a software project requires some local environment. So I guess I'm just looking for a bit of wiggle room here given that the community scopes from "I know what a container is and I'm comfortable using one" to "I created a GitHub account today to try to contribute through my web browser" and everything in between.

I guess I also want us to avoid going in circles for too long because this particular decision will never make everyone 100% happy due to the fact that there is an over-abundance of tools in this space. Everyone will have their favourite and won't have heard of 80% of the others. I, for instance, have no experience of hatch and so that wouldn't be the set of instructions that I would choose, just to provide an example of this :)

TL;DR is probably: I'm not explicitly against changing the installation guide, but let's also not overthink/over-engineer it.

**Comment by kallewesterling** [comment]
I agree with all that has been said above!

I do however want to point out something in this PR:

> TL;DR is probably: I'm not explicitly against changing the installation guide, but let's also not overthink/over-engineer it.

I haven't changed any instructions here, but only consolidated the instructions into the Community Handbook, as it is easier for folks to access and to give access to instructions for how to build the book locally, if someone wants to. I think a different PR can address the issues of changing things around in the local build instructions, like removing conda, etc. so someone (@JimMadge maybe?) should open a separate issue + PR addressing those issues.

**Comment by JimMadge** [comment]
@sgibson91 That is a great point. The most important things are that CI/deploy are the same and that CI gives contributors a preview and feedback on any errors.

In reality, people who are more likely to want to build and test locally are probably less likely to need instructions.

File: book/website/community-handbook/local-build.md (hunk 4 above)
Comment (JimMadge) [reply book/website/community-handbook/local-build.md:121]: I think it would be good to try and fix it here as we are improving that page in this PR and it's a bit of a bad look to pass over attribution and accessibility problems.

We can open an issue instead if it looks to be a more major task. What do you think @kallewesterling?

**Comment by kallewesterling** [comment]
> In reality, people who are more likely to want to build and test locally are probably less likely to need instructions.

I'd argue that it's still something that lowers barriers, and even those of us who are pretty fluent in these things (but aren't working day to day with it) still need some reminders every now and then :)

File: book/website/community-handbook/local-build.md (hunk 4 above)
Comment (kallewesterling) [reply book/website/community-handbook/local-build.md:121]: No, no! I'll just spend a few minutes on it now, to make sure it's rectified in this PR!

Comment (kallewesterling) [reply book/website/community-handbook/local-build.md]: Please have a look at https://github.com/the-turing-way/the-turing-way/pull/3446/commits/2e64e47f4672fe18c7da429002ad37e9fd654b48

**Comment by sgibson91** [comment]
> I haven't changed any instructions here

Yes, sorry. I meant generally, and in response to Jim's wonderings :)

**Review by JimMadge — APPROVED** [review]
LGTM
